pkg/services/publicdashboards/api/middleware.go

```diff
@@ -11,6 +15,31 @@ func SetPublicDashboardFlag() func(c *models.ReqContext) {
 	}
 }
 
+func RequiresValidAccessToken(publicDashboardService publicdashboards.Service) func(c *models.ReqContext) {
+	return func(c *models.ReqContext) {
+		accessToken, ok := web.Params(c.Req)[":accessToken"]
+
+		// Check access token is present on the request
+		if !ok || accessToken == "" {
+			c.JsonApiErr(http.StatusBadRequest, "Invalid access token", nil)
+			return
```

Comment:
This code is far enough away from the caller that I think it's worth leaving in. It can be tested in isolation and ensures that if it's mistakenly added to another route the proper behavior still happens.